modules/utils/include/hephaestus/utils/exception.h

```diff
@@ -62,11 +63,19 @@ constexpr void throwExceptionIf(bool condition, const std::string& message,
   }
 #else
   auto e = T{ message, location };
-  CHECK(!condition) << fmt::format("[ERROR {}] {} at {}:{}", e.what(), message, location.file_name(),
-                                   location.line());
+  ABSL_CHECK(!condition) << fmt::format("[ERROR {}] {} at {}:{}", e.what(), message, location.file_name(),
+                                        location.line());
 #endif
 }
 
+#ifdef DISABLE_EXCEPTIONS
+#define EXPECT_THROW_OR_CHECKFAIL(statement, expected_exception, expected_matcher)                           \
+  EXPECT_DEATH(statement, expected_matcher)
+#else
+#define EXPECT_THROW_OR_CHECKFAIL(statement, expected_exception, expected_matcher)                           \
+  EXPECT_THROW(statement, expected_exception)
+#endif
```

Comment:
Not sure if this can be done, but could we have:
```suggestion
#ifdef DISABLE_EXCEPTIONS
#define EXPECT_THROW(statement, expected_exception, expected_matcher)                           \
  EXPECT_DEATH(statement, expected_matcher)
#endif
```
in this way we do not need to refactor all the tests, but I am not sure if we can redefine a macro.

Reply:
Will check if we can overlay the gtest EXPECT_THROW with this one.

Reply:
overwriting did throw a warning that we do not allow.
changed it to EXPECT_THROW_OR_DEATH to be a bit more consistent with the underlying macro naming.